yt/data_objects/data_containers.py

```diff
@@ -413,12 +411,12 @@ def _generate_particle_field(self, field):
         return rv
 
     def _count_particles(self, ftype):
-        for (f1, f2), val in self.field_data.items():
+        for (f1, _f2), val in self.field_data.items():
             if f1 == ftype:
                 return val.size
         size = 0
```

Comment:
In retrospect, this may not be correct, but I do not have an immediate fix.  I *think* we probably want to return `val.shape[0]` which in most cases will be identical to `val.size` but in the case of vector or array fields will not.

Reply:
Let me know if I should just implement this.

Reply:
Let me know if I should just implement this.